flutter/lib/src/frames_tracking/span_frame_metrics_collector.dart

```diff
@@ -0,0 +1,75 @@
+// ignore_for_file: invalid_use_of_internal_member
+
+import 'package:meta/meta.dart';
+import '../../sentry_flutter.dart';
+import 'sentry_delayed_frames_tracker.dart';
+
+/// Collects frames from [SentryDelayedFramesTracker], calculates the metrics
+/// and attaches them to spans.
+@internal
+class SpanFrameMetricsCollector implements PerformanceContinuousCollector {
+  SpanFrameMetricsCollector(this._options, this._frameTracker);
+
+  final SentryFlutterOptions _options;
+  final SentryDelayedFramesTracker _frameTracker;
+
+  /// Stores the spans that are actively being tracked.
+  /// After the frames are calculated and stored in the span the span is removed from this list.
+  @visibleForTesting
+  final List<ISentrySpan> activeSpans = [];
+
+  @override
+  Future<void> onSpanStarted(ISentrySpan span) async {
+    return _tryCatch('onSpanFinished', () async {
+      if (span is NoOpSentrySpan) {
+        return;
+      }
+
+      activeSpans.add(span);
+      _frameTracker.resume();
+    });
+  }
+
+  @override
+  Future<void> onSpanFinished(ISentrySpan span, DateTime endTimestamp) async {
+    return _tryCatch('onSpanFinished', () async {
+      if (span is NoOpSentrySpan) {
+        return;
+      }
+
+      final startTimestamp = span.startTimestamp;
+      final metrics = _frameTracker.getFrameMetrics(
+          spanStartTimestamp: startTimestamp, spanEndTimestamp: endTimestamp);
+      metrics?.applyTo(span);
+
+      activeSpans.remove(span);
```

Comment:
we keep all frames in memory now, until all spans finish
instead, we can remove older frames when we finish a span, like [we did before](https://github.com/getsentry/sentry-dart/pull/2372/files#diff-1d0b11aa1a4e6d34ddbdb262ea2f0d33f1947ff6ccfbdc46f770e2ca92f08ab4L109-L110) 
```
          exceededFrames.removeWhere((frameTimestamp, _) =>
              frameTimestamp.isBefore(activeSpans.first.startTimestamp));
```

Reply:
I'm not sure whether adding this is wasting cpu power since this line removes frames that are the oldest / don't contribute anymore to any of the spans but since we only start frames tracking when we start and stop transactions there is no frame that can be remove.

for example stopping a child span would try to remove older frames but there is none older than the root span / transaction

or maybe my thinking here is wrong?

Reply:
that's true, until we run two (or more) transactions concurrently, right?
you could save the oldest frame timestamp, so that you can do just one comparison before deleting the frames
Something like 
```
if (finishedSpan.startDate > firstFrameStartTimestamp) {
  _delayedFrames.removeWhere((frame) => frame.startTimestamp.isBefore(spanStartTimestamp));
  firstFrameStartTimestamp = _delayedFrames.firstOrNull?.startTimestamp
}
```
wdyt?

Reply:
oh yeah sry didn't think of concurrent ones,  your suggestion is good 👍